src/emulator/functionsEmulator.ts

```diff
@@ -818,7 +832,129 @@ export class FunctionsEmulator implements EmulatorInstance {
     return process.execPath;
   }
 
-  invokeRuntime(frb: FunctionsRuntimeBundle, opts: InvokeRuntimeOpts): RuntimeWorker {
+  getUserEnvs(): Record<string, string> {
+    const projectInfo = {
+      functionsSource: this.args.functionsDir,
+      projectId: this.args.projectId,
+      isEmulator: true,
+    };
+
+    if (functionsEnv.hasUserEnvs(projectInfo)) {
+      try {
+        return functionsEnv.loadUserEnvs(projectInfo);
+      } catch (e) {
+        // Ignore - user envs are optional.
+        logger.debug("Failed to load local environment variables", e);
+      }
+    }
+    return {};
+  }
+
+  getSystemEnvs(triggerDef?: {
+    targetName: string;
+    triggerType: EmulatedTriggerType;
+  }): Record<string, string> {
+    const envs: Record<string, string> = {};
+
+    // Env vars guaranteed by GCF platform.
+    //   https://cloud.google.com/functions/docs/env-var
+    envs.GCLOUD_PROJECT = this.args.projectId;
+    envs.K_REVISION = "1";
+    envs.PORT = "80";
```

Comment:
This seems problematic

Reply:
We will clean this up once we refactor the functions emulator to stop using socket for comms. Is there another problem you are worried about?

Reply:
code may rely on PORT. At least in Run, you're responsible for listening on $PORT. When we're actually listening on a port this variable should be accurate.